src/lambda_handler.py

```diff
@@ -1,89 +1,231 @@
+        return await response.read()
+
+
+async def process_urls(cve_urls, db) -> None:
+    """Initialize Beanie ODM and begin processing CVE URLs."""
+    await init_beanie(database=motor_client[db], document_models=[CVEDoc])
+
+    # Create a new session for each URL to avoid throttling
+    async with aiohttp.ClientSession() as session:
+        tasks = []
+
+        for cve_url in cve_urls:
+            attempts = 0
+            retries = MAX_CVE_URL_RETRIES
 
-    return result
+            while True:
+                try:
+                    content = await fetch_url_content(session, cve_url)
+                    tasks.append(asyncio.create_task(unzip_and_process(content)))
+                    break
+                except aiohttp.ClientError as err:
+                    logging.debug(
+                        "Encountered a(n) %s exception while attempting to fetch URL '%s'",
+                        type(err).__name__,
+                        cve_url,
+                    )
+                    attempts += 1
+                    await asyncio.sleep(CVE_URL_RETRY_WAIT_SEC)
+                    if attempts <= retries:
+                        logging.warning(
+                            "Performing retry %d/%d for '%s'",
+                            attempts,
+                            retries,
+                            cve_url,
+                        )
+                    else:
+                        raise err
+    await asyncio.gather(*tasks)
 
 
-def handler(event, context) -> dict[str, Optional[str]]:
+def generate_urls():
+    """Return the NVD URLs for each year."""
+    current_year = datetime.utcnow().year
+    years = list(range(NVD_FIRST_YEAR, current_year + 1))
+    return [DEFAULT_NVD_URL.format(**{"year": year}) for year in years]
+
+
+def get_ssm_parameter(parameter_name: str) -> str:
+    """Retrieve the value of the provided SSM Parameter Store key."""
+    try:
+        resp = ssm_client.get_parameter(Name=parameter_name, WithDecryption=True)
+        return resp["Parameter"]["Value"]
+    except ClientError:
+        logging.error("Unable to retrieve SSM Parameter Store key: %s", parameter_name)
+        raise
+
+
+def build_mongodb_uri(
+    ssm_username: str, ssm_password: str, ssm_host: str, ssm_port: str, ssm_auth_db: str
+) -> Optional[str]:
+    """Build a MongoDB database URI from the provided SSM Parameter Store keys."""
+    uri = None
+    try:
+        # Ensure the username and password are safely encoded
+        username = urllib.parse.quote_plus(get_ssm_parameter(ssm_username))
+        password = urllib.parse.quote_plus(get_ssm_parameter(ssm_password))
+        host = get_ssm_parameter(ssm_host)
+        port = get_ssm_parameter(ssm_port)
+        auth_db = get_ssm_parameter(ssm_auth_db)
+        uri = f"mongodb://{username}:{password}@{host}:{port}/{auth_db}"
+    except ClientError as client_err:
+        logging.error("Unable to create MongoDB URI.")
+        logging.exception(client_err)
```

Comment:
Adjusting this since two values are no longer stored in SSM Parameter Store.

```suggestion
        # Ensure the database host and port were provided
        if not db_host:
            raise ValueError("The database hostname must be provided.")
        if not db_port:
            raise ValueError("The database port must be provided.")
        # Ensure the username and password are safely encoded
        username = urllib.parse.quote_plus(get_ssm_parameter(ssm_username))
        password = urllib.parse.quote_plus(get_ssm_parameter(ssm_password))
        auth_db = get_ssm_parameter(ssm_auth_db)
        uri = f"mongodb://{username}:{password}@{host}:{port}/{auth_db}"
    except (ClientError, ValueError) as err:
        logging.error("Unable to create MongoDB URI.")
        logging.exception(err)
```